moabb/evaluations/splitters.py

```diff
@@ -101,3 +110,164 @@ def split(self, y, metadata):
+        shuffle: bool = False,
+        random_state: int = None,
+        **cv_kwargs,
+    ):
+        self.cv_class = cv_class
+        self.cv_kwargs = cv_kwargs
+        self._cv_kwargs = dict(**cv_kwargs)
+
+        self.shuffle = shuffle
+        self.random_state = random_state
+
+        if not shuffle and random_state is not None:
+            raise ValueError("`random_state` should be None when `shuffle` is False")
+
+        params = inspect.signature(self.cv_class).parameters
+        # We need to check the random_state and shuffle parameters
+        # because for some cv_class, shuffle is not supported
+        # like ShuffleSplit or ShuffleGroupSplit.
+        if shuffle and ("shuffle" not in params and "random_state" not in params):
+            raise ValueError(
+                f"Shuffling is not supported for {cv_class.__name__}. "
+                "Choose a different `cv_class` or use `shuffle=False`."
+                "Example of `cv_class`: `GroupShuffleSplit`: "
+                "CrossSessionSplitter(shuffle=True, random_state=42, cv_class=GroupShuffleSplit)"
+            )
+
+        self._need_rng = "random_state" in params and (shuffle or "shuffle" not in params)
+
+        for p, v in [
+            ("shuffle", shuffle),
+        ]:
+            if p in params:
+                self._cv_kwargs[p] = v
+
+    def get_n_splits(self, metadata):
+        """
+        Return the number of splits for the cross-validation.
+
+        The number of splits is the number of subjects times the number of splits
+        of the inner cross-validation strategy.
+
+        We try to keep the same behaviour as the sklearn cross-validation classes.
+
+        Parameters
+        ----------
+        metadata: pd.DataFrame
+            The metadata containing the subject and session information.
+
+        Returns
+        -------
+        n_splits: int
+            The number of splits for the cross-validation
+        """
+        subjects = metadata["subject"].unique()
+        n_splits = 0
+        for subject in subjects:
+            subject_metadata = metadata.query("subject == @subject")
+            sessions = subject_metadata["session"].unique()
+
+            if len(sessions) <= 1:
+                continue  # Skip subjects with only one session
+
+            splitter = self.cv_class(**self._cv_kwargs)
+            n_splits += splitter.get_n_splits(
+                subject_metadata, groups=subject_metadata["session"]
+            )
+        return n_splits
+
+    def split(self, y, metadata):
+        # here, I am getting the index across all the subject
+        all_index = metadata.index.values
+        # I check how many subjects are here:
+        subjects = metadata["subject"].unique()
+
+        # To make sure that when I shuffle the subject, I shuffle the same way
+        # the session when the object is created
+        cv_kwargs = {**self.cv_kwargs}  # Copy the original kwargs
```

Comment:
You need to copy `_cv_kwargs` otherwise you won't have the `shuffle` argument.
```suggestion
        cv_kwargs = {**self._cv_kwargs}  # Copy the original kwargs
```